src/lib.rs

```diff
@@ -1551,6 +1557,44 @@ impl Builder {
     }
 }
 
+/// Setting for layout test generation.
+#[derive(Debug, Clone, PartialEq)]
+pub enum LayoutTests {
+    /// Include layout tests in the generated bindings. The default.
+    Emit,
+    /// Don't include any layout tests.
+    EmitNone,
+    /// Only emit the layout tests.
+    ///
+    /// The intended use case for this is for separating tests from the
+    /// generated bindings (as the tests are by nature non-portable, even if the
+    /// bindings otherwise would be).
+    ///
+    /// When used in this manner, you're encouraged to provide an an otherwise
+    /// identical set of options (even though many of them are effectively
+    /// ignored when this is set).
+    EmitOnly,
+}
+
+impl LayoutTests {
+    pub(crate) fn needed(&self) -> bool {
+        match self {
+            LayoutTests::Emit | LayoutTests::EmitOnly => true,
+            LayoutTests::EmitNone => false,
+        }
+    }
+}
+
+impl From<bool> for LayoutTests {
```

Comment:
This is mostly so that existing calls to `layout_tests` don't break. I also think it's sensible though.